### ISSUE TITLE
Add support for deploy token

### ISSUE THREAD

**Issue by giorgioazzinnaro** [xanzy/go-gitlab#801]
I use this client via the Terraform provider for GitLab, and have a use case where I'd like to be able to manage a deployment of certain images I have in my GitLab private image repository into Kubernetes.

In order to do this, I need a [deploy token](https://docs.gitlab.com/ee/user/project/deploy_tokens/) with `read_registry` permissions.

Looking at [GitLab API Reference](https://docs.gitlab.com/ee/api/api_resources.html) I was not able to find an endpoint that would be suitable to manage this...
However, it seems like this is just an undocumented feature: https://gitlab.com/gitlab-org/gitlab-foss/-/merge_requests/29864

A quick test proved this to be true.

Creating a deploy token:
```sh
curl -d name=test -d expires_at="" -d username=test -d scopes[]=read_registry -d https://gitlab.com/api/v4/projects/:project_id/deploy_tokens
```
```json
{
  "id": 149172,
  "name": "test",
  "username": "test",
  "expires_at": null,
  "scopes": [
    "read_registry"
  ],
  "token": "<redacted>"
}
```

And retrieving it:

```sh
curl  https://gitlab.com/api/v4/projects/:project_id/deploy_tokens
```
```json
[
  {
    "id": 149172,
    "name": "test",
    "username": "test",
    "expires_at": null,
    "scopes": [
      "read_registry"
    ]
  }
]
```

**Comment by giorgioazzinnaro** [comment]
A correction from my previous statement: it seems I just didn't look around well enough.

Here's the docs:
https://docs.gitlab.com/ce/api/deploy_tokens.html

I'll be opening a PR soon!

Cheers
Giorgio

**Comment by svanharmelen** [comment]
Cool, sounds good 👍

**Comment by giorgioazzinnaro** [comment]
I looked further into this... and I realised why I wasn't initially able to find the docs. Basically this is going to be released in version 12.9 in a couple of days. However, GitLab.com is already running this version, but the API is not entirely stable yet. So to anyone who'd like to use this, only GitLab.com works and there is still an issue with usernames.

See this for more details: https://gitlab.com/gitlab-org/gitlab/-/issues/21811
